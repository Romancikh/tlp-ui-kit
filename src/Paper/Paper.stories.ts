import type { Meta, StoryObj } from "@storybook/react";

import Paper from "./Paper";

const meta = {
  title: "Paper",
  component: Paper,
  parameters: {
    layout: "centered",
  },
  tags: ["autodocs"],
} satisfies Meta<typeof Paper>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {
    height: "128px",
    width: "128px",
    children: "Some content",
  },
};

export const Elevation: Story = {
  args: {
    height: "128px",
    width: "128px",
    children: "Some content",
    elevation: 12,
  },
};
