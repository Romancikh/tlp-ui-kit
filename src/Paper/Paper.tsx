import MuiPaper from "@mui/material/Paper";
import { PropsWithChildren } from "react";

type PaperProps = PropsWithChildren & {
  width?: "fit-content";
};

function Paper({ width, children }: PaperProps) {
  return <MuiPaper sx={{ width }}>{children}</MuiPaper>;
}

export default Paper;
