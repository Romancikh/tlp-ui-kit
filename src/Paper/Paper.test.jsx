import { render } from "@testing-library/react";
import Paper from "./Paper";

test("Paper renders correctly", () => {
  const { container } = render(
    <Paper height="128px" width="128px" children="Some content" />,
  );
  expect(container).toMatchSnapshot();
});
