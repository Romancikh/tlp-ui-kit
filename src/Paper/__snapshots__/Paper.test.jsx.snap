// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paper renders correctly 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1oj31o6-MuiPaper-root"
  >
    Some content
  </div>
</div>
`;
